src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -40,7 +41,7 @@ function HeatmapVis(): JSX.Element {
         aspectRatio={aspectRatio}
       >
         {/* Provide context again - https://github.com/react-spring/react-three-fiber/issues/262 */}
-        <HeatmapProvider {...props}>
+        <VisProviderForCanvas {...visProps}>
```

Comment:
As the provider does no longer depend on the vis, we could put `VisProviderForCanvas` inside the `VisCanvas` which would be quite convenient but could also render `VisCanvas` less independent.

Reply:
Done !